common/djangoapps/util/enterprise_helpers.py

```diff
@@ -71,6 +71,99 @@ def post_enterprise_course_enrollment(self, username, course_id, consent_granted
+                                "site": {
+                                    "domain": "example.com",
+                                    "name": "example.com"
+                                },
+                                "enable_data_sharing_consent": true,
+                                "enforce_data_sharing_consent": "at_login",
+                                "enterprise_customer_users": [
+                                    1
+                                ],
+                                "branding_configuration": {
+                                    "enterprise_customer": "cf246b88-d5f6-4908-a522-fc307e0b0c59",
+                                    "logo": "https://open.edx.org/sites/all/themes/edx_open/logo.png"
+                                },
+                                "enterprise_customer_entitlements": [
+                                    {
+                                        "enterprise_customer": "cf246b88-d5f6-4908-a522-fc307e0b0c59",
+                                        "entitlement_id": 69
+                                    }
+                                ]
+                            },
+                            "user_id": 5,
+                            "user": {
+                                "username": "staff",
+                                "first_name": "",
+                                "last_name": "",
+                                "email": "staff@example.com",
+                                "is_staff": true,
+                                "is_active": true,
+                                "date_joined": "2016-09-01T19:18:26.026495Z"
+                            },
+                            "data_sharing_consent": [
+                                {
+                                    "user": 1,
+                                    "state": "enabled",
+                                    "enabled": true
+                                }
+                            ]
+                        }
+                    ],
+                    "next": null,
+                    "start": 0,
+                    "previous": null
+                }
+            }
+
+        Raises:
+            ConnectionError: requests exception "ConnectionError", raised if if ecommerce is unable to connect
+                to enterprise api server.
+            SlumberBaseException: base slumber exception "SlumberBaseException", raised if API response contains
+                http error status like 4xx, 5xx etc.
+            Timeout: requests exception "Timeout", raised if enterprise API is taking too long for returning
+                a response. This exception is raised for both connection timeout and read timeout.
+
+        """
+        api_resource_name = 'enterprise-learner'
+        endpoint = getattr(self.client, api_resource_name)
+        querystring = {'username': user.username}
+
+        try:
+            response = endpoint().get(**querystring)
```

Comment:
I see no harm in adding caching to this call -- I think it is good protection against issues with service communication.